src/sources/kubernetes_logs/mod.rs

```diff
@@ -252,7 +252,9 @@ impl Source {
             max_read_bytes,
             // We want to use checkpoining mechanism, and resume from where we
             // left off.
-            start_at_beginning: false,
+            ignore_checkpoints: false,
+            // Match the default behavior
+            read_from: ReadFrom::default(),
```

Comment:
I think we should write a concrete value here, the one that matches the old behavior, so to simplify our choice of default for `file` source, although I do agree that they will probably end up being the same.

Reply:
I think this comes down to whether or not we'd want the k8s behavior to change if/when we change the file source default behavior. My assumption was yes, we do want it to match, which is why I centralized the definition of the default. We could make a decision to decouple the two defaults, but I'm not sure I see the value there. It's still very easy to change this value if we choose to.

Reply:
With an implicit change vs explicit change, I'd pick an explicit change in this case. There are practical reasons for it: explicit changes would be easier to review, preventing things go unnoticed.

Ideally, we'd want to get our E2E suite to a place where any change in behavior is caught - so this concern goes away, but it's practically impossible.

So, I'd say I'd be more comfortable with less dependency on the defaults at the code level specifically. It's fine to keep the behavior of the file source and kubernetes_logs source in-line, but it's better if the changes in the behavior are visible at the component source code level.